### PR TITLE
Stateless rest server order

### PR DESCRIPTION
**Pull request by qurben**
Based on #46

### CODE REVIEW

**Comment by gandreadis** [comment]
@qurben This is missing handling for order checkouts (as your new e2e test luckily found out 😅)... Looking into how to fix this atm

**Comment by gandreadis** [comment]
Alright, this is less trivial than expected. I can fix it by for now sending back a success event when the order checkout is received. This is of course not entirely correct (when the checkout fails, this is not reported back, and the response comes far quicker than the actual checkout happening). To really fix this, I think we need to make `Rest` in `orders` stateful, too, so that it has a sense of outstanding orders. Does any of you see a better approach?

**Comment by qurben** [comment]
Right, so I think that for now it is enough to keep it as it was implemented previously. (fast return)

The thing with state is an interesting one. And I think also a problem which is very deeply rooted in how this is implemented. For each level that you do a new 'request' on kafka you'd need to keep track of which request you are handling. And what should be returned to the user.

Maybe another way to look at this is to say already in the rest server that when doing a checkout that the request should return when a 'CheckoutSuccess' or 'CheckoutFailed' with a specific id is encountered withing the timeout. I am however not entirely sure if this is doable for all requests.

Another way would be to embrace the 'eventual consistency' thing and say doing a checkout will return an order which is 'in processing' and then you can check if your order is finished.

**Comment by qurben** [comment]
Not sure why creating an order hangs it seems that messages are not correctly delivered to the orders service, while they are properly delivered to other services.

File: docker-compose.yml [hunk 1]
```diff
@@ -11,24 +11,13 @@ services:
...
-    image: nginx:latest
-    volumes:
-      - ./nginx/nginx.conf:/etc/nginx/nginx.conf
-    links:
-      - orders
-      - payments
-      - stock
-      - rest
-    ports:
-      - "8080:80"
```
Comment (gandreadis) [inline docker-compose.yml]: Does the removal of this mean we can also remove the nginx.conf file?

**Comment by qurben** [comment]
Yes!

**Comment by gandreadis** [comment]
I think I've found a fix, @qurben - previously, all except for the user service had the same group ID in their Kafka config. When I made them all unique (so each service its own ID), it seemed to work on my machine! Let's see whether CI agrees

**Comment by gandreadis** [comment]
🎉 Yup, it works :D Removing the nginx.conf file now and then we're ready to merge

**Comment by qurben** [comment]
Epic!

**Pull request merged**